openhands/llm/llm.py

```diff
@@ -236,11 +212,11 @@ def wrapper(*args, **kwargs):
 
             # log the response
             message_back = resp['choices'][0]['message']['content']
+            if message_back:
+                llm_response_logger.debug(message_back)
 
-            llm_response_logger.debug(message_back)
-
-            # post-process to log costs
-            self._post_completion(resp)
+                # post-process to log costs
+                self._post_completion(resp)
```

Comment:
I think even if the `message_back` is empty string - we might still incur costs (eg input tokens), right? I'd suggest we move this line out of the if statement.

```suggestion
            self._post_completion(resp)
```

Reply:
Here's why I put that in on purpose: I saw integration tests having 2 completions, but 3 steps and thus fail

Reply:
If that's the case, let's leave this as is then... (until it cause some error in the future)